src/esssans/beam_center_finder.py

```diff
@@ -227,15 +229,11 @@ def minimize(
 
 
 def beam_center(
-    data: sc.DataArray,
-    data_monitors: Dict[str, sc.DataArray],
-    direct_monitors: Dict[str, sc.DataArray],
-    wavelength_bins: sc.Variable,
-    q_bins: Union[int, sc.Variable],
-    gravity: bool = False,
-    minimizer: str = 'Nelder-Mead',
-    tolerance: float = 0.1,
-) -> Tuple[sc.Variable, sc.Variable]:
+    data: MaskedData[SampleRun],
+    graph: ElasticCoordTransformGraph,
+    wavelength_bins: WavelengthBins,
+    norm: Clean[SampleRun, Denominator],
```

Comment:
@nvaytet By using this, we will automatically get the denominator including the direct-beam factor, which was not used in the old implementation. If I remember correctly the argument was that we need the beam-center-find for the direct-beam iteration. However, `DirectBeam` is an optional dependency in the graph, so I think this should just work...

... unless there is an explicit reason why we should *not* include the direct-beam factor here?

Reply:
> we need the beam-center-find for the direct-beam iteration

correct

Reply:
... but it is fine to use it if we have it?

Reply:
Yes, and I would actually argue that if it is pixel dependent, then it is probably required?
I think we dropped it in the sans2d case because we assumed the direct beam was just wavelength dependent, and we were trying to overlap 4 curves. Including/omitting the direct beam would affect the shape of all 4 curves in the same manner, and we can thus drop it (I hope I didn't mess something up in my logic).

However, if it is now pixel dependent, then probably we cannot since different quadrants will be affected differently?

Maybe this is our chicken-egg problem then, because the direct beam computation requires the beam center finding and vice versa?

Reply:
You can still run it without. The `BeamCenter` is optional in the `esssans` pipeline, just as it was optional in `ess.sans`.

Reply:
I think I wasn't really asking whether you can run the pipeline or not, but more conceptually, how are we going to compute the direct beam in practise?

Even without thinking about `sciline`, if you want to compute a pixel-dependent direct beam, you need to find the beam center. But to compute that, you need the direct beam. Do you just have a first guess for the beam center?

Reply:
> Even without thinking about sciline, if you want to compute a pixel-dependent direct beam, you need to find the beam center. But to compute that, you need the direct beam. Do you just have a first guess for the beam center?

Didn't you prove with your work on `ess.sans.beam_center_finter` that the direct beam is *not* required for finding the beam center?

Reply:
That was my conclusion, as long as the direct beam is not pixel-dependent...